src/Calculator.ts

```diff
@@ -345,3 +339,196 @@ function addSDE(report: R4.IMeasureReport, measureURL: string, result: Execution
     });
   });
 }
+
+function popForResult(pr: PopulationResult) {
+  const pop = <R4.IMeasureReport_Population>{};
+  pop.code = {
+    coding: [
+      {
+        system: 'http://terminology.hl7.org/CodeSystem/measure-population',
+        code: pr.populationType,
+        display: POPULATION_DISPLAY_MAP[pr.populationType]
+      }
+    ]
+  };
+  pop.count = pr.result ? 1 : 0;
+  return pop;
+}
+
+// // Code->Display https://terminology.hl7.org/1.0.0/CodeSystem-measure-population.html
+const POPULATION_DISPLAY_MAP = {
+  [PopulationType.IPP]: 'Initial Population',
+  [PopulationType.DENOM]: 'Denominator',
+  [PopulationType.DENEX]: 'Denominator Exclusion',
+  [PopulationType.DENEXCEP]: 'Denominator Exception',
+  [PopulationType.NUMER]: 'Numerator',
+  [PopulationType.NUMEX]: 'Numerator Exclusion',
+  [PopulationType.MSRPOPL]: 'Measure Population',
+  [PopulationType.MSRPOPLEX]: 'Measure Population Exclusion',
+  [PopulationType.OBSERV]: 'Measure Observation'
+};
+
+// http://build.fhir.org/ig/HL7/cqf-measures/ValueSet-aggregate-method.html
+function aggregate(aggregation: string, observations: number[]) {
+  // Note: add break for any non-returning case
+  switch (aggregation) {
+    case AggregationType.SUM:
+      // sum	Sum	The measure score is determined by adding together the observations derived from the measure population.
+      return observations.reduce((accumulator, currentValue) => accumulator + currentValue);
+    case AggregationType.AVERAGE:
+      // average	Average	The measure score is determined by taking the average of the observations derived from the measure population.
+      return observations.reduce((accumulator, currentValue) => accumulator + currentValue) / observations.length;
+    case AggregationType.MEDIAN:
+      // median	Median	The measure score is determined by taking the median of the observations derived from the measure population.
+      return median(observations);
+    case AggregationType.MIN:
+      // minimum	Minimum	The measure score is determined by taking the minimum of the observations derived from the measure population.
+      return Math.min(...observations);
+    case AggregationType.MAX:
+      // maximum	Maximum	The measure score is determined by taking the maximum of the observations derived from the measure population.
+      return Math.max(...observations);
+    case AggregationType.COUNT:
+      // count	Count	The measure score is determined as the number of observations derived from the measure population.
+      return observations.length;
+    default:
+      // should not default, TODO: add error?
+      return observations.length;
```

Comment:
You can add an error throw here if you want. It'll be caught by a try/catch upstream. E.g.:

``` Javascript
throw new Error(`Measure score aggregation type ${aggregation} not supported`);
```

Reply:
Love it.